tiledb/sm/query/readers/reader_base.cc

```diff
@@ -145,6 +147,134 @@ bool ReaderBase::skip_field(
 /*        PROTECTED METHODS       */
 /* ****************************** */
 
+std::vector<uint64_t> ReaderBase::tile_offset_sizes() {
```

Comment:
Note to reviewers: This was move almost as is from SparseIndexReaderBase. I have only added two "if (!dense) {".